cvmfs/cache_ram.h

```diff
@@ -269,12 +269,22 @@ class RamCacheManager : public CacheManager {
     return open_fds_[fd].handle != kInvalidHandle;
   }
 
+  inline MemoryKvStore *GetStore(const ReadOnlyFd &fd) {
+    if (fd.is_volatile) {
+      return &volatile_entries_;
+    } else {
+      return &regular_entries_;
+    }
+  }
+
   int AddFd(const ReadOnlyFd &fd);
   int64_t CommitToKvStore(Transaction *transaction);
   virtual int DoOpen(const shash::Any &id);
 
   uint64_t max_size_;
+  size_t fd_pivot_;
   std::vector<ReadOnlyFd> open_fds_;
+  std::vector<int> fd_index_;
```

Comment:
Could this be a vector of `unsigned`?  It would remove the need for one or two `assert`s and clarify that contents can never be negative.